presto-main/src/main/java/com/facebook/presto/operator/aggregation/AccumulatorCompiler.java

```diff
@@ -360,7 +360,9 @@ else if (parameterType == Slice.class) {
                         expressions.add(index.invoke("getSlice", Slice.class, getChannel, position));
                     }
                     else if (parameterType == Block.class) {
-                        expressions.add(index.invoke("getObject", Block.class, getChannel, position));
+                        // Even though the method signature requires a Block parameter, we can pass an Object here.
+                        // A runtime check will assert that the Object passed as a parameter is actually of type Block.
+                        expressions.add(index.invoke("getObject", Object.class, getChannel, position));
```

Comment:
Can you add an explanation to the commit message describing why this is needed? I.e., what causes the query to fail, etc.